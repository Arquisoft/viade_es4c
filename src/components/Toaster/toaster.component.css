.toaster{
  display:
          flex;
  flex-direction: column;
  justify-content: space-evenly;
  flex: 1 1 auto;}

.toaster > span {
  font-weight: bold;
  text-transform: uppercase;
}
.toaster > p {
  margin: 0;
  padding: 2px 0;
  color: inherit;
}
.toaster > a {
  color: white;
  display: block;
  margin: 5px 0;
}