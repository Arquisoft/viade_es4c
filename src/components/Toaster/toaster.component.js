import React from "react";

type Props = {
  title: String,
  content: String,
  link: { href: string, label: string }
};


const Toaster = (props: Props) => {
  const { title, content, link } = props;
  return (
    <div className="toaster">
      <span>{title}</span>
      <p>{content}</p>
      {link && (
        <a href={link.href} target="_blank" rel="noopener noreferrer">
          {link.label}
        </a>
      )}
    </div>
  );
};

export default Toaster;
