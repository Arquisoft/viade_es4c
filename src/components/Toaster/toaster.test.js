import React from 'react';
import { shallow } from 'enzyme';
import Toaster from "./toaster.component";

describe('Toaster', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Toaster debug />);
  
    expect(component).toMatchSnapshot();
  });
});
