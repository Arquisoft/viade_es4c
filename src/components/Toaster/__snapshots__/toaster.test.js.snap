// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster should render correctly in "debug" mode 1`] = `
<div
  className="toaster"
>
  <span />
  <p />
</div>
`;
